src/ocean/meta/traits/Basic.d

```diff
@@ -0,0 +1,424 @@
+        || is( Unqual!(T) == real );
+}
+
+///
+unittest
+{
+    static assert ( isRealType!(double));
+    static assert (!isRealType!(long));
+    static assert (!isRealType!(cdouble));
+}
+
+
+/*******************************************************************************
+
+    Params:
+        T = type to check
+
+    Returns:
+        `true` if `T` is a complex floating point type
+
+*******************************************************************************/
+
+public template isComplexType( T )
+{
+    const bool isComplexType =
+           is( Unqual!(T) == cfloat )
+        || is( Unqual!(T) == cdouble )
+        || is( Unqual!(T) == creal );
+}
+
+///
+unittest
+{
+    static assert ( isComplexType!(cdouble));
+    static assert ( isComplexType!(Const!(cdouble)));
+    static assert (!isComplexType!(double));
+}
+
+/*******************************************************************************
+
+    Params:
+        T = type to check
+
+    Returns:
+        `true` if `T` is an imaginary floating point type
+
+*******************************************************************************/
+
+public template isImaginaryType( T )
+{
+    const bool isImaginaryType =
+           is( Unqual!(T) == ifloat )
+        || is( Unqual!(T) == idouble )
+        || is( Unqual!(T) == ireal );
+}
+
+///
+unittest
+{
+    static assert ( isImaginaryType!(idouble));
+    static assert ( isImaginaryType!(Const!(idouble)));
+    static assert (!isImaginaryType!(double));
+}
+
+/*******************************************************************************
+
+    Params:
+        T = type to check
+
+    Returns:
+        `true` if `T` is any floating-point type: real, complex, or
+        imaginary.
+
+*******************************************************************************/
+
+public template isFloatingPointType( T )
+{
+    const bool isFloatingPointType =
+           isRealType!(T)
+        || isComplexType!(T)
+        || isImaginaryType!(T);
+}
+
+///
+unittest
+{
+    static assert (isFloatingPointType!(double));
+    static assert (isFloatingPointType!(ifloat));
+}
+
+/*******************************************************************************
+
+    Used by other traits to distinguish between dynamic and static arrays
+    instead of plain `bool` values.
+
+*******************************************************************************/
+
+public enum ArrayKind
+{
+    NotArray    = 0,
+    Static      = 1,
+    Dynamic     = 2,
+    Associative = 3,
+}
+
+/*******************************************************************************
+
+    Check if type is an array type and which kind of array it is
+
+    Params:
+        T = type to check
+
+    Returns:
+        `ArrayKind` value indicating if `T` is an array and if it is, static or
+        dynamic
+
+*******************************************************************************/
+
+public template isArrayType ( T )
+{
+    static if (is(T U == U[]))
```

Comment:
Depends on kind of the bug. If bug is unlikely to be fixed and workaround complicates code in many places, I'd totally consider relying on it like I did here. Rationale is that if it is going to be fixed, such intrusive change would require deprecations anyway and we will be able to adjust. If it is not going to be fixed, we pollute the code for no reason.